sycl/test/basic_tests/image_accessor_types.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: not %clangxx -fsyntax-only %s 2>&1 | FileCheck %s
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+class mod_image;
+int main() {
+  half4 src[16];
+  image<2> srcImage(src, image_channel_order::rgba, image_channel_type::fp16,
+                    range<2>(4, 4));
+  queue myQueue;
+  myQueue.submit([&](handler &cgh) {
+    accessor<float, 2, access::mode::read, access::target::image> NotValidType1(
+        srcImage, cgh);
+    // CHECK: The data type of an image accessor must be only cl_int4, cl_uint4, cl_float4 or cl_half4
+    accessor<int2, 2, access::mode::read, access::target::image> NotValidType2(
+        srcImage, cgh);
+    // CHECK: The data type of an image accessor must be only cl_int4, cl_uint4, cl_float4 or cl_half4
+    accessor<float4, 2, access::mode::read, access::target::image>
+        ValidSYCLFloat(srcImage, cgh);
+    accessor<int4, 2, access::mode::read, access::target::image> ValidSYCLInt(
```

Comment:
Just to understand the code, why are you adding the int4/uint4/half4/float4 tests?
They serve no purpose and have no control on the test result.
Are they here so that anyone who is looking at the test case can see what the valid constructor will look like?

Reply:
I added all valid types for the future, in case someone breaks something in accessor constructor.